php/libraries/Candidate.class.inc

```diff
@@ -98,7 +98,8 @@ class Candidate
 
         if (!is_array($row) || sizeof($row) == 0) {
             throw new LorisException(
-                "Could not select Candidate data from the database (DCCID: $candID)",
+                'Could not select Candidate data from the database (DCCID: '
+                                                . htmlspecialchars($candID),
```

Comment:
Yes, it would break in the situations where the output isn't going directly to a web browser. tools/scripts, cron jobs, ajax scripts which send output which are rendered to react components (which do their own escaping, likely causing it to be double escaped), etc.

I don't know of anywhere in LORIS outside of main.php that catches exceptions right now, so I don't think it should be difficult to do exhaustively.